### PR TITLE
BUG: pd.crosstab(s1, s2) handle column index incorrectly when both series have tuple names

### LINKED ISSUE #18321: pd.crosstab(s1, s2) keeps dummy MultiIndex as columns if both s1 and s2 have tuple name
#### Code Sample, a copy-pastable example if possible

```python
In [2]: s1 = pd.Series(range(3), name=('a', 'b'))

In [3]: s2 = pd.Series(range(3), name=('c', 'd'))

In [4]: pd.crosstab(s1, s2)
Out[4]: 
                   
('c', 'd')  0  1  2
(a, b)             
0           1  0  0
1           0  1  0
2           0  0  1

In [5]: list(pd.crosstab(s1, s2).columns)
Out[5]: [('', 0), ('', 1), ('', 2)]
```

#### Problem description

While the example is similar to that reported in #18304, the fix doesn't seem to be related. Reminder: test by expanding the test in #18309 .

#### Expected Output

``` python
In [5]: list(pd.crosstab(s1, s2).columns)
Out[5]: [0, 1, 2]
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: 9c799e2c4331e5e42cfda03323eef165feb5be1a
python: 3.5.3.final.0
python-bits: 64
OS: Linux
OS-release: 4.9.0-3-amd64
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: it_IT.UTF-8
LOCALE: it_IT.UTF-8

pandas: 0.22.0.dev0+135.g9c799e2c4
pytest: 3.2.3
pip: 9.0.1
setuptools: 36.7.0
Cython: 0.25.2
numpy: 1.12.1
scipy: 0.19.0
pyarrow: None
xarray: None
IPython: 6.2.1
sphinx: 1.5.6
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: 1.2.0dev
tables: 3.3.0
numexpr: 2.6.1
feather: 0.3.1
matplotlib: 2.0.0
openpyxl: None
xlrd: 1.0.0
xlwt: 1.1.2
xlsxwriter: 0.9.6
lxml: None
bs4: 4.5.3
html5lib: 0.999999999
sqlalchemy: 1.0.15
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: 0.2.1

</details>

### PR DESCRIPTION
**Pull request by charlesdong1991**
- [ ] closes #18321
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/reshape/test_pivot.py [hunk 1]
```diff
@@ -2549,6 +2549,53 @@ def test_crosstab_tuple_name(self, names):
         result = pd.crosstab(s1, s2)
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.parametrize(
+        "s1_data, s1_name, s2_data, s2_name, "
+        "expected_index, expected_column, expected_data",
```
Comment (WillAyd) [inline pandas/tests/reshape/test_pivot.py]: what is the unique feature between these sets of parameters you are trying to test? Wasn't immediately clear to me; may be OK to do without it if it simplifies test

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/tests/reshape/test_pivot.py (hunk 1 above)
Comment (charlesdong1991) [reply pandas/tests/reshape/test_pivot.py]: yeah, removed. the unique feature was different shape of output, and with one of `index` or `column` having same value, like `[1,1,1]`, but probably made it unclear. I now just used the example described in the issue

File: pandas/core/reshape/pivot.py [hunk 2]
```diff
@@ -596,6 +598,11 @@ def crosstab(
         **kwargs,
     )
 
+    # Remove extra level from `[__dummy__]` pivoting
```
Comment (jreback) [inline pandas/core/reshape/pivot.py]: can you give a comment on why you are doing this.

**Review by jreback — APPROVED** [review]
lgtm. minor comment, pls rebase as well.

**Comment by charlesdong1991** [comment]
ping @jreback

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @charlesdong1991

**Pull request merged**